src/utilities/helper.jl

```diff
@@ -73,3 +73,63 @@ function reconstruct!(r, d::MultivariateDistribution, val::AbstractVector, n::In
     r .= val
     return r
 end
+
+struct FlattenIterator{Tname, Tvalue}
+    name::Tname
+    value::Tvalue
+end
+
+Base.length(iter::FlattenIterator) where {T} = _length(iter.value)
+@inline _length(a::AbstractArray) = sum(_length, a)
+@inline _length(::Number) = 1
+
+Base.eltype(iter::FlattenIterator{String}) = Tuple{String, _eltype(typeof(iter.value))}
+@inline _eltype(::Type{TA}) where {T, TA <: AbstractArray{T}} = _eltype(T)
```

Comment:
Again, is `@inline` needed? And it seems `T` is not needed?
```suggestion
_eltype(::Type{TA}) where {TA <: AbstractArray} = _eltype(eltype(TA))
```